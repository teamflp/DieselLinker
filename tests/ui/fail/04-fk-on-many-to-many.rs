use diesel_linker::relation;

#[derive(Debug)]
struct Post;

#[derive(Debug)]
#[relation(
    relation_type = "many_to_many",
    model = "Post",
    fk = "user_id", // This is not used for a many_to_many relation
    join_table = "users_posts",
    fk_parent = "user_id",
    fk_child = "post_id",
    backend = "sqlite"
)]
struct User;

fn main() {}
