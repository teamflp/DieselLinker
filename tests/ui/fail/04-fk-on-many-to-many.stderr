error: `fk` is not used for `many_to_many` relations. Use `fk_parent` and `fk_child` instead.
  --> tests/ui/fail/04-fk-on-many-to-many.rs:10:5
   |
10 |     fk = "user_id", // This is not used for a many_to_many relation
   |     ^^

error[E0412]: cannot find type `User` in this scope
 --> tests/ui/fail/04-fk-on-many-to-many.rs:6:10
  |
6 | #[derive(Debug)]
  |          ^^^^^ not found in this scope
